### ISSUE TITLE
New release to support py3.10

### ISSUE THREAD

**Issue by jeffkala** [arista-eosplus/pyeapi#246]
develop branch has a fix required for py3.10 to work. https://github.com/arista-eosplus/pyeapi/blob/236503162d1aa3ecc953678ec05380f1f605be02/pyeapi/api/abstract.py#L44

Is there plans for a release soon?

**Comment by dlyssenko** [comment]
Hi @jeffkala, we're working slowly towards a new release (but there's still lots of work, especially for system tests). I'm aiming for the new release in the first half of the next year, in meanwhile please use latest `develop` branch.

**Comment by jeffkala** [comment]
Okay, unfortunately `develop` branch works in a development environment, but the application i'm trying to release to pypi can't have a direct dependency.

```
WARNING  Error during upload. Retry with the --verbose option for more details. 
ERROR    HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/        
         Invalid value for requires_dist. Error: Can't have direct dependency:  
         'pyeapi @ git+https://github.com/arista-eosplus/pyeapi.git@develop'    
```
looks like I'll have to drop py3.10 support in my app for the time being.  Thanks for the information.  I'll keep an eye on this issue for updates on the release date.

**Comment by jvanderaa** [comment]
Please also look at supporting 3.11 that dropped recently.

**Comment by dlyssenko** [comment]
yes, will add coverage for 3.11, and sorry for the inconvenience delaying the release - pyeapi is an OS project and its support is a best effort, but I'm making a progress towards the new release every week now.

**Comment by jeffkala** [comment]
Thanks and understood on OS projects!

**Comment by jeffkala** [comment]
@dlyssenko as far as the next release is there a milestone I'm not seeing? I'd love to do some PRs on issues that are required to help push this through.

**Comment by dlyssenko** [comment]
@jeffkala, I'm setting up 3 milestones:
1. clean up all the reported bugs and PRs
2. Since pyeapi was integrated into EOS (4.26.0) there are some system tests which fail (actually a big part of those) - need to fix them all
3. Since we (at Arista) are now migrating from Python2 over to Python3 - some system tests also started failing - need to fix those too
After that it'll be time for the next release (and update of the integrated pyeapi version in EOS).

**Comment by jeffkala** [comment]
Happy to try and help out in this, few questions below would help me get started.
- Is the thought to move to something like python-poetry for dependency management? Or sticking with setup.py.
- Dropping all py2.7, py3.4-3.6 support? Since all of those are deprecated at this point?
- Drop `netaddr` and just use standard `ipaddress` from the standard lib?

**Comment by ryanmerolle** [comment]
how goes this?

**Comment by dlyssenko** [comment]
the new pyeapi release 1.0.0 is out (with Python 3 only support). Closing this question.

**Comment by jeffkala** [comment]
> the new pyeapi release 1.0.0 is out (with Python 3 only support). Closing this question.

Thanks @dlyssenko for getting this out.
